unison-cli/src/Unison/CommandLine/InputPatterns.hs

```diff
@@ -1165,8 +1167,26 @@ loadPullRequest =
 parseUri :: String -> String -> Either (P.Pretty P.ColorText) ReadRemoteNamespace
 parseUri label input = do
   first
-    (fromString . show) -- turn any parsing errors into a Pretty.
+    prettyPrint
     (P.parse UriParser.repoPath label (Text.pack input))
+  where
+    prettyPrint :: P.ParseError Char Void -> P.Pretty P.ColorText
+    prettyPrint (P.TrivialError sp _ _) =
+      let col = (P.unPos $ P.sourceColumn $ NE.head sp) - 1
+          row = (P.unPos $ P.sourceLine $ NE.head sp) - 1
+          errorLine = lines input !! row
+      in P.lines ["I couldn't parse the repository address given above.",
+               P.newline,
+               P.string errorLine,
+               P.string $ replicate col ' ' <> "^-- This is where I gave up."
+              ]
+    prettyPrint (P.FancyError sp _) =
+      let col = (P.unPos $ P.sourceColumn $ NE.head sp) - 1
+      in P.lines ["I couldn't parse the repository address given above.",
+               P.newline,
+               P.string input,
```

Comment:
I notice in the TrivialError case you need to grab the appropriate row, do we not need to do that for FancyErrors too?
I suspect we'll usually be parsing URI's independently as a single line, but we should still handle that case if we can.

It seems possible to unify these two implementations into a single one which just takes the span right? Or am I missing something there?